### DISCUSSION TITLE
Where is config the emmm... _build/ port?

### DISCUSSION THREAD

**Discussion by PeterlitsZo** [solidjs/solid-start#1355] [Q&A]
When I run `bun run dev`, it will start a dev mode server on port 3000. And it is OK when I using the VSCode Remote - it will proxy it.

But the question is that it also open another port to handle those `_build/` requests:

<img width="411" alt="截屏2024-02-29 14 13 43" src="https://github.com/solidjs/solid-start/assets/25707198/ffcbb0be-4c02-4724-8066-dd6c4343ce84">

The problem is that the port is always changed and I have to change my proxy config every time when I try to restart the dev mode server.

Do you guys know where is the place to change its behave (I try myself to dive into it to patch it but failed) or to config it (which is better).

**Comment by knpwrs** [comment]
It looks to me like this is configured with `vite.server.hmr.port` in `app.config.js`:

```ts
export default defineConfig({
  vite: {
    server: {
      hmr: {
        port: 4444,
      },
    },
  },
});
```

**Reply by web-dev-sayantan** [reply]
This solution is creating a strange error(I made sure that nothing is open on that port).
My app.config.ts looks like this:
```typescript
import { defineConfig } from "@solidjs/start/config";

export default defineConfig({
  vite: {
    server: {
      hmr: {
        port: 4001,
      },
    },
  },
});
```
<img width="750" alt="image" src="https://github.com/user-attachments/assets/c2319e83-01c1-41e0-824a-a4ea9e57b7b0">

**Reply by elliotnash** [reply]
@Ishanz23 I was having the same issue before I found this post https://www.gandalf-pro.dev/blog/solid-start-hmr-with-docker

Basically vinxi starts vite for the client, the server, and for "use server" functions -- you need to set a different port for each one

```ts
import { defineConfig } from "@solidjs/start/config";

const hmrPorts = {
  client: 4440,
  server: 4441,
  'server-function': 4442,
}

export default defineConfig({
  vite: ({ router }) => ({
    server: {
      hmr: {
        port: hmrPorts[router]
      },
    }
  })
});
```

**Reply by solisadeqi** [reply]
facing same issue!!
